lib/lita/external/robot.rb

```diff
@@ -1,9 +1,26 @@
+require 'zk'
+
 module Lita
   module External
     class Robot < ::Lita::Robot
+
+      def initialize_leader_lock
+        return unless ENV['ZOOKEEPER_PEERS']
+        zk_peers = ENV['ZOOKEEPER_PEERS']
+        zk_lock = "lita_#{config.robot.name}_#{ENV['ENV']}"
+        @locker = ZK::Locker.exclusive_locker(ZK.new(zk_peers), zk_lock)
+        @locker.lock(wait: false)
+      end
```

Comment:
Right. Now that you mention it. Checking env like that is a bit invasive. Also `ENV['ENV']`, is not a Lita thing, it's purely a Shopify convention.

I think we could simply use the `:master_loaded` event to set the `locker` on the `robot`. That way `lita-external` is back to be pretty much agnostic. And you can implement the same thing with a Redis lock if you want.